idaes/ui/fsvis/static/js/paper.js

```diff
@@ -66,6 +66,42 @@ export class Paper {
         // context menu
         self._paper.on("element:contextmenu", function(cellView, evt) {
             cellView.model.rotate(90)
+            // This is needed to keep the text labels for the unit models in the correct orientation
+            // x and y were specifically picked to keep the label in the same place 
+            // in relation to the unit model (bottom middle)
+            // TODO Make this figuring out the x and y positions a function so that we can compute it
+            const angle = cellView.model.angle()
+            switch (angle) {
```

Comment:
So let me suggest a more data-driven way to approach this.
```
function translate_for_angle(angle, width, height) {
   // TODO: replace with geometry that considers width and height
   const angle_translation = {0: [0, 5], 90: [38, -35], 180: [0, -72], 270: [-38, -34]};
   return angle_translation[angle];
}
// then in this function:
[x,y] = translate_for_angle(angle);
```
This hides your heuristic somewhere that you can change later.
Is the case of a number not in the list normal behavior? That seems like it should be a runtime error (here it's  a stack trace, but you could explicitly catch it).

Reply:
Your implementation makes much more sense. Thank you

No it should always be one of those numbers because that's the only rotation we allow. But I did add a catch and error message for that.